genderize.py

```diff
@@ -0,0 +1,184 @@
+                                    names = [name.replace('+','-'), name.replace('+',' '), name.replace('+','').capitalize()]
+                                else:
+                                    names = [name]
+                                for name in names:
+                                    if name not in self.genderDict:
+                                        self.genderDict[name] = {}
+                                        self.genderDict[name][mf] = {}                                   
+                                    else:  
+                                        if mf not in self.genderDict[name]:
+                                            self.genderDict[name][mf] = {} 
+                                            
+                                    for country,freq in zip(self.countries,frequencies):
+                                        if freq != ' ':
+                                            norm_freq = int((float(self.pop[country]) * 1000) * 0.02 * (2 ** (int(freq,16) - 10)))
+                                            #norm_freq = int(((float(pop[country]))*1000) * (math.log(int(freq,16))))
+                                            # print(int(float(pop[country]) * 1000)) 
+                                            # print((math.log(int(freq,16),2)))
+                                            
+                                            self.genderDict[name][mf][country] = norm_freq
+               
+            self.count += 1
+        
+        for [name, frequencies] in self.shortNames:
+            shortName, longName = name.split()
+            if shortName in self.genderDict and not longName in self.genderDict:
+                for nameList in self.genderDict[shortName]:
+                    if longName in self.genderDict:
+                        self.genderDict[longName].append(nameList)
+                    else:
+                        self.genderDict[longName] = [nameList]
+                    
+            elif longName in self.genderDict and not shortName in self.genderDict:
+                for nameList in self.genderDict[longName]:
+                    if shortName in self.genderDict:
+                        self.genderDict[shortName].append(nameList)
+                    else:
+                        self.genderDict[shortName] = [nameList]
+                    
+        #Sorts names for male and female with values 
+                        
+        for key in self.genderDict.keys():
+                if 'M' in self.genderDict[key]:
+                    self.listmale.append(key) 
+                else:
+                    if 'F' in self.genderDict[key]:
+                        self.listfem.append(key) 
+                      
+        self.all_names = [(i, 'm') for i in self.listmale] + [(i, 'f') for i in self.listfem]
+    
+    
+    #loop to generate data for classifier
+    
+    #while val <= len(all_names) * 0.8:
+    
+        for x in range(0, 1):
```

Comment:
This should not be init--move this to an appropriately named helper function, e.g. kfoldTrainer() or train() or buildModel(), etc. Then add a visualize() or report() function that produces the charts and evaluation measures on demand. This can still all be part of the Genderize class, but don't clutter init, and don't do more than is necessary to initialize the gender classifier here. In general, this evaluation training stuff should not happen when we are using the class to classify unseen samples--only when we want to create a new model. Otherwise, Genderizer should simply read a previously created model file (saved with pickle) and be ready to classify.